src/pipecat/services/openai.py

```diff
@@ -126,7 +156,13 @@ async def _process_context(self, context: OpenAILLMContext):
                 tool_call = chunk.choices[0].delta.tool_calls[0]
                 if tool_call.function and tool_call.function.name:
                     function_name += tool_call.function.name
-                    # yield LLMFunctionStartFrame(function_name=tool_call.function.name)
+                    tool_call_id = tool_call.id
+                    # only send a function start frame if we're not handling the function call
+                    if function_name in self._callbacks.keys():
+                        if function_name in self._start_callbacks.keys():
+                            await self._start_callbacks[function_name](self)
+                    else:
+                        await self.push_frame(LLMFunctionStartFrame(function_name=tool_call.function.name, tool_call_id=tool_call_id))
```

Comment:
I believe we should get rid of `LLMFunctionStartFrame` and `LLMFunctionCallFrame` for now. It seems we don't need them in the ported patient-intake. We can always re-add them if we really find a use-case that can only be solved this way.